ext/nnstreamer/extra/nnstreamer_python3_helper.cc

```diff
@@ -397,72 +398,101 @@ int addToSysPath (const gchar *path)
 int
 parseTensorsInfo (PyObject *result, GstTensorsInfo *info)
 {
-  int ret = 0;
+  guint i, j, rank;
+
   if (PyList_Size (result) < 0)
     return -1;
 
   info->num_tensors = PyList_Size (result);
-  for (guint i = 0; i < info->num_tensors; i++) {
+  for (i = 0; i < info->num_tensors; i++) {
     /** don't own the reference */
     PyObject *tensor_shape = PyList_GetItem (result, (Py_ssize_t)i);
     if (nullptr == tensor_shape) {
-      Py_ERRMSG ("parseTensorsInfo() has failed (1).");
-      return -1;
-    }
-
-    PyObject *shape_dims = PyObject_CallMethod (tensor_shape, (char *)"getDims", NULL);
-    if (nullptr == shape_dims) {
-      Py_ERRMSG ("parseTensorsInfo() has failed (2).");
+      Py_ERRMSG ("parseTensorsInfo() has failed, cannot get TensorShape object.");
```

Comment:
How about using ```__FUNCTION__``` or ```__PRETTY_FUNCTION__``` for better portability?

Reply:
updated